lib/hanami/model/sql/entity/schema.rb

```diff
@@ -38,6 +38,24 @@ def initialize(registry, relation, mapping)
             freeze
           end
 
+          # Process attributes
+          #
+          # @param attributes [#to_hash] the attributes hash
+          #
+          # @raise [TypeError] if the process fails
+          #
+          # @since 1.0.1
+          # @api private
+          def call(attributes)
+            Utils::Hash.new(
+              schema.call(attributes)
+            ).deep_symbolize!
```

Comment:
Before this change `Sql::Entity::Schema#call` was inherited from `Entity::Schema` which had this exact implementation. Since I changed the latter but did not want to change the behaviour of the former, I added this to override the inherited `Entity::Schema#call` method. I.e., for inferred schema the behaviour should be as it was before.

I did not ponder how inferred schemas treat associations at all. 🤔  Maybe we should first discuss the desired behaviour for all the cases?